src/context/layers/impact.ts

```diff
@@ -168,76 +183,134 @@ export async function fetchImpactLayerData(
     );
   }
 
-  // Calculate a bounding box for each feature that we have baseline data for
-  const matchingFeatures = adminBoundaries.features.reduce(
-    (acc, f) => {
-      const feature = f as GeoJsonBoundary;
-      const baseline =
-        featureIntersectsImage(feature, image) &&
-        getBaselineDataForFeature(feature, baselineData);
-      return baseline
-        ? acc.concat({
-            id: baseline.adminKey,
-            baseline,
-            feature,
-            bounds: bbox(feature),
-          })
-        : acc;
-    },
-    [] as {
-      id: string;
-      baseline: BaselineRecord;
-      feature: GeoJsonBoundary;
-      bounds: ReturnType<typeof bbox>;
-    }[],
-  );
+  // eslint-disable-next-line fp/no-mutation
+  let activeFeatures: GeoJsonBoundary[];
+  if (layer.api) {
```

Comment:
I feel like giant `if` statements like this generally make the code more difficult to follow/reason about (which is another reason that `fp/no-mutation` exists). I'd break this out into separate utility functions:

```typescript
const activeFeatures = layer.api ? loadFeaturesFromApi(layer, ...) : loadFeaturesFromClient(...)
```